tests/it/src/commands/create_diff_cases.rs

```diff
@@ -0,0 +1,129 @@
+            sliders.lines().count()
+        );
+
+        let sliders: HashSet<_> = sliders
+            .lines()
+            .take(count)
+            .map(|line| {
+                let parts: Vec<_> = line.split_ascii_whitespace().collect();
+
+                match parts[..] {
+                    [before, after, ..] => (before, after),
+                    _ => todo!(),
+                }
+            })
+            .collect();
+
+        eprintln!("{sliders:?}");
+
+        let repo = gix::open(worktree_dir)?;
+
+        let asset_dir = asset_dir.unwrap_or("assets".into());
+        let assets = destination_dir.join(asset_dir.to_os_str()?);
+
+        eprintln!("{prefix} create directory '{assets}'", assets = assets.display());
+        if !dry_run {
+            std::fs::create_dir_all(&assets)?;
+        }
+
+        let mut buf = Vec::new();
+
+        let script_name = "make_diff_for_sliders_repo.sh";
+
+        let mut blocks: Vec<String> = vec![format!(
+            r#"#!/usr/bin/env bash
+# TODO:
+# `git diff --no-index` returns 1 when there's differences, but 1 is treated as an error by the
+# shell.
+# set -eu -o pipefail
+
+ROOT="$(cd "$(dirname "${{BASH_SOURCE[0]}}")" && pwd)"
+
+git init
+echo .gitignore >> .gitignore
+echo {asset_dir}/ >> .gitignore
+echo {script_name} >> .gitignore
+
+mkdir -p {asset_dir}
+"#
+        )];
+
+        for (before, after) in sliders.iter() {
+            let revspec = repo.rev_parse(*before)?;
+            let old_blob_id = revspec
+                .single()
+                .context(format!("rev-spec '{before}' must resolve to a single object"))?;
+            let (old_path, _) = revspec
+                .path_and_mode()
+                .context(format!("rev-spec '{before}' must contain a path"))?;
+
+            let revspec = repo.rev_parse(*after)?;
+            let new_blob_id = revspec
+                .single()
+                .context(format!("rev-spec '{after}' must resolve to a single object"))?;
+            let (new_path, _) = revspec
+                .path_and_mode()
+                .context(format!("rev-spec '{after}' must contain a path"))?;
+
+            eprintln!("{old_blob_id:?} {old_path:?} {new_blob_id:?} {new_path:?}");
+
+            let dst_old_blob = assets.join(format!("{old_blob_id}.commit"));
+            let dst_new_blob = assets.join(format!("{new_blob_id}.commit"));
+            if !dry_run {
+                let old_blob = repo.objects.find_blob(&old_blob_id, &mut buf)?.data;
+                std::fs::write(dst_old_blob, old_blob)?;
+
+                let new_blob = repo.objects.find_blob(&new_blob_id, &mut buf)?.data;
+                std::fs::write(dst_new_blob, new_blob)?;
+            }
+
+            blocks.push(format!(
+                r#"git diff --no-index "$ROOT/{asset_dir}/{old_blob_id}.commit" "$ROOT/{asset_dir}/{new_blob_id}.commit" > .git/{old_blob_id}-{new_blob_id}.baseline
+cp "$ROOT/{asset_dir}/{old_blob_id}.commit" assets/
```

Comment:
Are these `.commit`s?

Reply:
Thanks for the suggestion, changed to `.blob`.